nethttp/client.go

```diff
@@ -159,7 +161,15 @@ func (h *Tracer) clientTrace() *httptrace.ClientTrace {
 }
 
 func (h *Tracer) getConn(hostPort string) {
-	ext.HTTPUrl.Set(h.sp, hostPort)
+	host, portString, err := net.SplitHostPort(hostPort)
+	if err != nil {
+		ext.PeerHostname.Set(h.sp, hostPort)
+	} else {
+		ext.PeerHostname.Set(h.sp, host)
+		if port, err := strconv.Atoi(portString); err != nil {
+			ext.PeerPort.Set(h.sp, uint16(port))
+		}
+	}
```

Comment:
actually, I think both of these are wrong. The url and peer tags should be set for all traced http requests (i.e. in RoundTrip method), not only those for which `httptrace.ClientTrace` is enabled. What should really happen here is:
```
h.sp.LogKV(log.String("event", "GetConn"), log.String("hostPort", hostPort))
```
(or split host/port into two k/v pairs if not lazy).

I don't know how much more overhead `httptrace.ClientTrace` adds, it would be nice to make it optional. The `RoundTrip` is really what's needed for the actual tracing/propagation.